controlplane/api/v1beta1/rke2controlplane_types.go

```diff
@@ -28,7 +28,7 @@ import (
 
 const (
 	// RKE2ControlPlaneFinalizer allows the controller to clean up resources on delete.
-	RKE2ControlPlaneFinalizer = "rke2.controleplane.cluster.x-k8s.io"
+	RKE2ControlPlaneFinalizer = "rke2.controlplane.cluster.x-k8s.io"
```

Comment:
This is an interesting one. Users will potentially have clusters that are using the old finalizer name with the typo and if we change it like this the existing finalizer will never be removed  

I think its safer if we do something like this:


```
RKE2ControlPlaneFinalizer = "rke2.controlplane.cluster.x-k8s.io"

RKE2ControlPlaneLegacyFinalizer = "rke2.controleplane.cluster.x-k8s.io"
```

And then change controller to always remove the legacy finalizer. Lets chat about this.

Reply:
Added suggestion with legacy finalizer and modified `rke2controlplane_controller` to remove the old one if it's still attached to the cluster.